rotkehlchen/chain/ethereum/decoding/decoder.py

```diff
@@ -352,6 +377,15 @@ def _maybe_decode_simple_transactions(
                 counterparty='gas',
             ))
 
+        # Decode internal transactions after gas so gas is always 0 indexed
+        self._maybe_decode_internal_transactions(
```

Comment:
No no no. You already decode internal transactions up there. You should not do that in 2 places.

What you need to do instead is move the gas burn detection code where you have the comment you did not remove. `# Check for gas spent`